File: import/new/2785.go
// https://leetcode.com/problems/sort-vowels-in-a-string
var vowels = []byte("AEIOUaeiou")
var idx [128]int8

func init() {
    for i, c := range vowels { idx[c] = int8(i) + 1 }
}

func sortVowels(s string) string {
    b := []byte(s)

	var counts [10]int
	for _, c := range b {
		k := idx[c] - 1
		if k >= 0 { counts[k]++ }
	}

	res := make([]byte, len(b))
	copy(res, b)

	cur := 0
	for i, c := range b {
		k := idx[c] - 1
		if k >= 0 {
			for counts[cur] == 0 { cur++ }
			res[i] = vowels[cur]
			counts[cur]--
		}
	}
	return string(res)
}

File: import/new/2785.py
# https://leetcode.com/problems/sort-vowels-in-a-string
class Solution:
    def sortVowels(self, s: str) -> str:
        vowels = "AEIOUaeiou"
        counts = Counter(s)
        for v in vowels:
            s = s.replace(v, '_')
        for v in vowels:
            cnt = counts.get(v, 0)
            if cnt:
                s = s.replace('_', v, cnt)
        return s
